modules/plugin/imagemosaic/src/main/java/org/geotools/gce/imagemosaic/catalog/CogConfigurationBean.java

```diff
@@ -0,0 +1,145 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ *
+ *    (C) 2020, Open Source Geospatial Foundation (OSGeo)
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+package org.geotools.gce.imagemosaic.catalog;
+
+import it.geosolutions.imageioimpl.plugins.cog.CogUri;
+import java.lang.reflect.InvocationTargetException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import org.apache.commons.beanutils.BeanUtils;
+import org.geotools.gce.imagemosaic.Utils;
+import org.geotools.gce.imagemosaic.catalog.index.Indexer;
+import org.geotools.gce.imagemosaic.catalog.index.IndexerUtils;
+import org.geotools.util.Utilities;
+
+/** Bean containing all COG related configuration properties */
+public class CogConfigurationBean {
```

Comment:
Can we please just call this a "CogConfiguration"? Nowhere else in GeoTools we use this "Bean" suffix convention, nowhere in Java own platform either.

Reply:
will do.

Reply:
Done